tasks/vendoring/__init__.py

```diff
@@ -137,6 +154,105 @@ def vendor(ctx, vendor_dir):
         apply_patch(ctx, patch)
 
 
+def download_licenses(ctx, vendor_dir):
+    log('Downloading licenses')
+    tmp_dir = vendor_dir / '__tmp__'
+    ctx.run(
+        'pip download -r {0}/vendor.txt --no-binary :all: -d {1}'.format(
+            str(vendor_dir),
+            str(tmp_dir),
+        )
+    )
+    for sdist in tmp_dir.iterdir():
+        extract_license(vendor_dir, sdist)
+    drop_dir(tmp_dir)
+
+
+def extract_license(vendor_dir, sdist):
+    if sdist.suffixes[-2] == '.tar':
+        ext = sdist.suffixes[-1][1:]
+        with tarfile.open(sdist, mode='r:{}'.format(ext)) as tar:
+            found = find_and_extract_license(vendor_dir, tar, tar.getmembers())
+    elif sdist.suffixes[-1] == '.zip':
+        with zipfile.ZipFile(sdist) as zip:
+            found = find_and_extract_license(vendor_dir, zip, zip.infolist())
+    else:
+        raise NotImplementedError('new sdist type!')
+
+    if not found:
+        log('License not found in {}, will download'.format(sdist.name))
+        license_fallback(vendor_dir, sdist.name)
+
+
+def find_and_extract_license(vendor_dir, tar, members):
+    found = False
+    for member in members:
+        try:
+            name = member.name
+        except AttributeError:  # zipfile
+            name = member.filename
+        if 'LICENSE' in name or 'COPYING' in name:
+            if '/test' in name:
+                # some testing licenses in hml5lib and distlib
+                log('Ignoring {}'.format(name))
+                continue
+            found = True
+            extract_license_mamber(vendor_dir, tar, member, name)
+    return found
+
+
+def license_fallback(vendor_dir, sdist_name):
+    """Hardcoded license URLs. Check when updating if those are still needed"""
+    for libname, url in HARDCODED_LICENSE_URLS.items():
+        if libname in sdist_name:
+            *_, name = url.rpartition('/')
```

Comment:
`_, _, name = url.rpartition('/')` is Python 2 compatible.

Reply:
It is a Python 3 only script, but we don't exclude it from the Python 2 linter run, for better or worse. It's no big deal - I actually think the explicit version is clearer, anyway.

Reply:
It's not a big deal, I agree.